pkg/limatmpl/embed.go

```diff
@@ -552,11 +555,68 @@ func (tmpl *Template) combineNetworks() {
 	}
 }
 
+// yamlfmt will fail with a buffer overflow while trying to retain line breaks if the line
+// is longer than 64K. We will encode all text files that have a line that comes close.
+const maxLineLength = 65000
+
+// Break base64 strings into shorter chunks. Technically we could use maxLineLength here,
+// but shorter lines look better.
+const base64ChunkLength = 76
+
+// binaryString returns a base64 encoded version of the binary string, broken into chunks
+// of at most base64ChunkLength characters per line.
+func binaryString(s string) string {
+	encoded := base64.StdEncoding.EncodeToString([]byte(s))
+	if len(encoded) <= base64ChunkLength {
+		return encoded
+	}
+
+	// Estimate capacity: encoded length + number of newlines
+	lineCount := (len(encoded) + base64ChunkLength - 1) / base64ChunkLength
+	builder := strings.Builder{}
+	builder.Grow(len(encoded) + lineCount)
+
+	for i := 0; i < len(encoded); i += base64ChunkLength {
+		end := i + base64ChunkLength
+		if end > len(encoded) {
+			end = len(encoded)
+		}
+		builder.WriteString(encoded[i:end])
+		builder.WriteByte('\n')
+	}
+
+	return builder.String()
+}
+
 // updateScript replaces a "file" property with the actual script and then renames the field to newName ("script" or "content").
-func (tmpl *Template) updateScript(field string, idx int, newName, script string) {
+func (tmpl *Template) updateScript(field string, idx int, newName, script, file string) {
+	start := 0
+	line := 1
+	encodeReason := ""
+	for i, r := range script {
+		if !(unicode.IsPrint(r) || r == '\n' || r == '\r' || r == '\t') {
+			encodeReason = fmt.Sprintf("unprintable character %q at offset %d", r, i)
+			break
+		}
+		if i-start > maxLineLength {
+			encodeReason = fmt.Sprintf("line %d (offset %d) is longer than %d characters", line, start, maxLineLength)
```

Comment:
It would be great to cover this line with a unit test.

Reply:
I did this by changing the package name for `embed_test.go` from `limatmpl_test` to just `limatmpl`. Otherwise I would have to export `encodeScriptReason`. Neither one is desirable, but changing the test package name seems slightly better.

BTW, I know the `i-start` condition is off by 1 or 2 (depending on if you include the newline in the max line length), but since the limit is kind of arbitrary, and well below the hard limit that starts breaking, I don't think it matters. Hopefully nobody has text files with such long lines anyways.